components/style/values/specified/mod.rs

```diff
@@ -345,14 +357,15 @@ impl Time {
         from_calc: bool)
         -> Result<Time, ()>
     {
-        let seconds = match_ignore_ascii_case! { unit,
-            "s" => value,
-            "ms" => value / 1000.0,
+        let (seconds, unit) = match_ignore_ascii_case! { unit,
+            "s" => (value, TimeUnit::Second),
+            "ms" => (value / 1000.0, TimeUnit::Millisecond),
```

Comment:
If we want to do that, we probably want to make the value wrapped by an enum like many other value types, so that we wouldn't mix values from different units by mistake. That would probably need more changes which doesn't seem to be worthy to me...

This way also makes the struct more compact (8 bytes vs. 12 bytes), although that's probably not a big deal either.